src/components/slider/attributes.js

```diff
@@ -0,0 +1,203 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var fontAttrs = require('../../plots/font_attributes');
+var colorAttrs = require('../color/attributes');
+var extendFlat = require('../../lib/extend').extendFlat;
+
+var stepsAttrs = {
+    _isLinkedToArray: true,
+
+    method: {
+        valType: 'enumerated',
+        values: ['restyle', 'relayout', 'animate', 'update'],
+        dflt: 'restyle',
+        role: 'info',
+        description: [
+            'Sets the Plotly method to be called on click.'
+        ].join(' ')
+    },
+    args: {
+        valType: 'info_array',
+        role: 'info',
+        freeLength: true,
+        items: [
+            { valType: 'any' },
+            { valType: 'any' },
+            { valType: 'any' }
+        ],
+        description: [
+            'Sets the arguments values to be passed to the Plotly',
+            'method set in `method` on click.'
+        ].join(' ')
+    },
+    label: {
+        valType: 'string',
+        role: 'info',
+        dflt: '',
+        description: 'Sets the text label to appear on the slider'
+    }
+};
+
+module.exports = {
+    visible: {
+        valType: 'boolean',
+        role: 'info',
+        dflt: true,
+        description: [
+            'Determines whether or not the slider is visible.'
+        ].join(' ')
+    },
+
+    active: {
+        valType: 'string',
+        role: 'info',
+        min: -1,
+        dflt: 0,
+        description: [
+            'Determines which button (by index starting from 0) is',
+            'considered active.'
+        ].join(' ')
+    },
+
+    steps: stepsAttrs,
+
+    updateevent: {
+        valType: 'string',
+        arrayOk: true,
+        role: 'info',
+        description: [
+            'The name of the event to which this component subscribes',
+            'in order to trigger updates. When the event is received',
+            'the component will attempt to update the slider position',
+            'to reflect the value passed as the data property of the',
+            'event. The corresponding step\'s API method is assumed to',
+            'have been triggered externally and so is not triggered again',
+            'when the event is received. If an array is provided, multiple',
+            'events will be subscribed to for updates.'
+        ].join(' ')
+    },
+
+    updatevalue: {
+        valType: 'string',
+        arrayOk: true,
+        role: 'info',
+        description: [
+            'The property of the event data that is matched to a slider',
+            'value when an event of type `updateevent` is received. If',
+            'undefined, the data argument itself is used. If a string,',
+            'that property is used, and if a string with dots, e.g.',
+            '`item.0.label`, then `data[\'item\'][0][\'label\']` will',
```

Comment:
Ok. `Plotly.restyle` and `Plotly.relayout` use use `[]` to delimit array items in an attribute string e.g. `item[0].label` via [`Lib.nestedProperty`](https://github.com/plotly/plotly.js/blob/708e0ffd9ee827fd2640334a072f86605dcb2f5d/src/lib/nested_property.js#L29) .

I believe the `item.0.label` is a little more clear. But, maybe we should stick to `item[0].label` for consistency?